infrastructure/testdriver/click_iframe.html

```diff
@@ -0,0 +1,25 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>TestDriver click method</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/resources/testdriver.js"></script>
+<script src="/resources/testdriver-vendor.js"></script>
+
+<iframe src="click_child.html"></iframe>
+
+<script>
+async_test(t => {
```

Comment:
Optional, but this (and the other tests) could be written as a promise test which might make the failure case clearer:

```
promise_test(async t => {
  await new Promise(resolve => { addEventListener("load", resolve); });
  let child = frames[0];
  let button = child.document.getElementById("button");
  await test_driver.click(button);
  assert_equals(child.document.getElementById("log").textContent, "PASS");
});
```

For me its a cleaner syntax (but that's obviously subjective), but it also reduces the error handling required. (Thrown errors or rejected promises should get a full stack trace).

Reply:
I made everything single page tests instead which also removes most of the boilerplate. idk which is better overall but I'd say they're at least comparable.